src/manifolds/MetricManifold.jl

```diff
@@ -58,13 +59,14 @@ where $g_{ij,k}=\frac{∂}{∂ p^k} g_{ij}$ is the coordinate
 derivative of the local representation of the metric tensor. The dimensions of
 the resulting multi-dimensional array are ordered $(i,j,k)$.
 """
-christoffel_symbols_first(::MetricManifold, ::Any)
+christoffel_symbols_first(::MetricManifold, B::AbstractBasis, ::Any)
```

Comment:
Our basis-related functions all put the basis argument as last. Should we do that here as well for consistency? A bonus would be that it can then have a default value.

Reply:
That's a good idea, I'll change order of arguments. What would be the right default value? Just `DefaultOrthonormalBasis()`? Another route would be through `get_default_atlas`, `get_chart_index` and `induced_basis`.

Reply:
> What would be the right default value? Just `DefaultOrthonormalBasis()`?

I think `DefaultBasis()`? Because there's no need for the basis to be orthonormal or even orthogonal, it should just ideally be fast, and `DefaultBasis` is meant to be the fastest. Besides, it forwards to `DefaultOrthogonalBasis` and `DefaultOrthonormalBasis` anyways.

> Another route would be through `get_default_atlas`, `get_chart_index` and `induced_basis`.

Can you elaborate on this route?

Reply:
I think `DefaultBasis` is primarily supposed to be the fastest at `get_vector` and `get_coordinates`, which doesn't necessarily correspond to being the fastest at computing coefficients of higher order tensors.

Anyway, this question touches an important point, specifically if one wants to specify a metric tensor, which basis are they supposed to do it in? The examples are currently for induced bases but that can be changed. You can take a look at the "scaled Sphere metric" test case where I take the default atlas for the sphere (`get_default_atlas`), then for a given point `p` I call `get_chart_index` to get the index of the chart from the atlas that is most suitable for representing neighborhood of `p`, and then I work in the basis induced by that chart (`induced_basis`).

IIRC a basis of the tangent space is insufficient to express coefficients of things like Christoffel symbols, unless it's an induced basis in which case you also have an atlas. Right now there is one underspecified piece here, that is I do not strictly enforce the basis of what vector space has to be passed to metric functions. An earlier design here just had atlas and chart parameters instead of a basis parameter but later I found it somewhat inflexible so I generalized it. Still, the core idea is that you would usually pass something that indicates a chart to metric-related functions.